### PR TITLE
Store all connection information in class DbConnection

### PR DESCRIPTION
**Pull request by JoanneBogart**
Addresses issue #54
Simplifies code in a number of places; largely transparent to users.

### CODE REVIEW

File: src/dataregistry/query.py [hunk 1]
```diff
@@ -85,32 +85,28 @@ class Query:
...
+        db_connection : DbConnection object
+            Encompasses sqlalchemy engine, dialect (database backend)
+            and schema version
         """
-        self._engine = db_engine
-        self._dialect = dialect
-        if dialect == "sqlite":
+        self._engine = db_connection.engine
+        self._dialect = db_connection.dialect
+        if self._dialect == "sqlite":
```
Comment (stuartmcalpine) [inline src/dataregistry/query.py:101]: Is this needed? Is this not repeating what is done when the DbConnection class is created?

Just need `self._schema_version = db_connection.schema`?

File: src/cli/query.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import os
-from dataregistry.db_basic import SCHEMA_VERSION, create_db_engine
```
Comment (stuartmcalpine) [inline src/cli/query.py:2]: Dont need to import SCHEMA_VERSION anymore?

File: src/cli/query.py [hunk 3]
```diff
@@ -22,10 +22,10 @@ def dregs_ls(owner, owner_type, show_all, dregs_config):
     """
 
     # Establish connection to database
-    engine, dialect = create_db_engine(config_file=dregs_config)
-
+    #engine, dialect = create_db_engine(config_file=dregs_config)
```
Comment (stuartmcalpine) [inline src/cli/query.py:25]: can delete this line?

File: src/dataregistry/DREGS.py [hunk 4]
```diff
@@ -46,22 +46,21 @@ def __init__(
 
         # If no schema specified, go to the default one
         if schema_version is None:
-            self.schema_version = SCHEMA_VERSION
-        else:
-            self.schema_version = schema_version
+            schema_version = SCHEMA_VERSION
```
Comment (stuartmcalpine) [inline src/dataregistry/DREGS.py:49]: Can just pass `schema_version` from the `__init__` straight through to `DbConnection`, as it's init also defaults it to `SCHEMA_VERSION` if None?

i.e., dont need these lines anymore

        # If no schema specified, go to the default one
        if schema_version is None:
            schema_version = SCHEMA_VERSION

File: src/dataregistry/DREGS.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import os
 
-from dataregistry.db_basic import SCHEMA_VERSION, create_db_engine
```
Comment (stuartmcalpine) [inline src/dataregistry/DREGS.py:3]: can remove `SCHEMA_VERSION` from import if comment below ok

File: src/dataregistry/registrar.py [hunk 6]
```diff
@@ -26,29 +26,25 @@
 class Registrar:
     def __init__(
         self,
-        db_engine,
-        dialect,
+        db_connection,
         owner=None,
         owner_type=None,
-        schema_version=SCHEMA_VERSION,
```
Comment (stuartmcalpine) [inline src/dataregistry/registrar.py:33]: can remove `SCHEMA_VERSION` import

**Review by stuartmcalpine — CHANGES_REQUESTED** [review]
looks good, just few comments

File: src/dataregistry/query.py (hunk 1 above)
Comment (JoanneBogart) [reply src/dataregistry/query.py:101]: Right.

File: src/cli/query.py (hunk 2 above)
Comment (JoanneBogart) [reply src/cli/query.py:2]: No, I guess not.

**Review by stuartmcalpine — APPROVED** [review]

**Comment by stuartmcalpine** [comment]
looks good to me

**Pull request merged**